File: app/src/main/java/com/example/myapplication/api/models/LoginResponse.java
package com.example.myapplication.api.models;

public class LoginResponse {
    private boolean success;
    private String error;
    private Data data;

    public boolean isSuccess() { return success; }
    public String getError() { return error; }
    public Data getData() { return data; }

    public static class Data {
        private String userId;
        private String firstName;
        private String lastName;
        private String email;
        private String role;
        private String token;
        private String roleSpecificId;

        public String getUserId() { return userId; }
        public String getFirstName() { return firstName; }
        public String getLastName() { return lastName; }
        public String getEmail() { return email; }
        public String getRole() { return role; }
        public String getToken() { return token; }
        public String getRoleSpecificId() { return roleSpecificId; }
    }
}

File: ClinicAPI/src/API/DTOs/LoginResponse.cs
namespace API.DTOs;

public class LoginResponse
{
    public bool Success { get; set; }
    public UserData? Data { get; set; }
    public string? Error { get; set; }
    public string? Message { get; set; }
}

public class UserData
{
    public string UserId { get; set; } = string.Empty;
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public string Token { get; set; } = string.Empty;
    public string RoleSpecificId { get; set; } = string.Empty; // PatientId, DoctorId, or ClerkId
}
